src/vehicle.cpp

```diff
@@ -4693,7 +4794,12 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
                 }
             }
         } else if( ret.type == veh_coll_body ) {
-            int dam = obj_dmg*dmg_mod/100;
+            int dam = obj_dmg * dmg_mod / 100;
+
+            if( is_body_collision && parts[ret.part].enabled &&
```

Comment:
I'm not sure if it's better to keep checking of ```parts[ret.part].enabled && part_info( ret.part ).has_flag( "CHAINSAW" )``` here of move it inside ```damage_from_chainsaw()``` function.